src/QirRuntime/lib/TracingContext/TracingContext.cpp

```diff
@@ -0,0 +1,103 @@
+#include <algorithm>
+#include <assert.h>
+#include <iostream>
+
+#include "QAPI.hpp"
+#include "ResourceStatistics.hpp"
+
+using namespace quantum;
+ResourceStatistics g_stats;
+
+void QAPI_DestroyContext(QuantumExecutionContext* context) // NOLINT
+{
+    assert(context == nullptr);
+
+    std::ostringstream os;
+    g_stats.Print(os);
+    std::cout << os.str();
+    g_stats = {0};
+}
+
+// Qubit management
+EXPORTQAPI Qubit QAPI_AllocateQubit(QuantumExecutionContext* context) // NOLINT

```

Comment:
For parameters that are not referenced inside functions, if available it would be great to use a macro like `UNREFERENCED_PARAMETER` if available to express intent.